### ISSUE TITLE
Convex decomposition leads to inaccurate shape for simple meshes

### ISSUE THREAD

**Issue by BlenderSleuth** [godotengine/godot#29971]
**Godot version:** 05a0a68c72cc16c443301398ab93e8d838401ac0

**OS/device including version:**
Windows 10 x86_64

**Issue description:**
Using the old quickhull method, when using the Mesh->Create Convex Collision Sibling, a simple scaled cube mesh would produce a perfect convex hull collision shape. Now, with the new convex decomposition, the same simple cube mesh produces an inaccurate shape:
![shape](https://user-images.githubusercontent.com/11659668/59957943-0b066900-94e2-11e9-9ccf-4ffba715eb2a.PNG)

**Steps to reproduce:**
Import this gltf file to a godot project:  [cube.zip](https://github.com/godotengine/godot/files/3316424/cube.zip)
Use Mesh->Create Convex Collision Sibling(s)
See that the collision shape does not line up with the mesh instance (use wireframe mode for a better look)

**Minimal reproduction project:**
[test.zip](https://github.com/godotengine/godot/files/3316426/test.zip)

**Comment by jitspoe** [comment]
I'm having the same problem when importing gltf meshes that have "convcolonly" on them.  In 3.1, these would import correctly.  In 3.2, the import alters the vertex positions slightly.

3.1: 
![image](https://user-images.githubusercontent.com/7751182/77384806-b471d800-6d5c-11ea-9e81-4e8de9a3b72a.png)
Point values are exactly the same as they are in Blender.

3.2: 
![image](https://user-images.githubusercontent.com/7751182/77384851-d23f3d00-6d5c-11ea-9786-a1c8bdbd1818.png)

Notice how the top and bottom are broken into triangles instead of being flat, and the point values are slightly off.  Example: y of 0 becomes 0.094 and 0.047 (strangely halfway between 0.094 and 0.0).

**Comment by akien-mga** [comment]
I checked and the regression is fairly old, as it's already present in https://downloads.tuxfamily.org/godotengine/3.2/alpha0-unofficial/ (a34b77e271da5f3d550acfac1e5557f2467c6266).

**Comment by akien-mga** [comment]
This is likely caused by 5823b5d77d47339c3c981953ca8178e7ef14b6df, CC @reduz.

**Comment by Calinou** [comment]
I guess we should have separate import flags for the old (QuickHull-based) and new (VHACD-based) convex collision shape creation.

**Comment by jitspoe** [comment]
Nah, the algorithm should be fixed to not create bad collision with simple shapes.

**Comment by Calinou** [comment]
@jitspoe Unfortunately, [VHACD is basically unmaintained](https://github.com/kmammou/v-hacd) and there are no alternatives we know of.

**Comment by akien-mga** [comment]
>  Unfortunately, VHACD is basically unmaintained and there are no alternatives we know of.

Well that doesn't mean that we can't fix VHACD ourselves :)

**Comment by jitspoe** [comment]
So it seems VHACD is for massive, complex models.  Would it make sense to just have it kick in on > 200 verts or something, otherwise use the quick one we used previously?  Or maybe you explicitly pick which one you want to use from the mesh menu and importing meshes with -convcolonly uses the old quick hull method (since, if you're explicitly making convex collision meshes in blender, you have probably already simplified them) and the other collision options that used to make triangle meshes instead use VHACD?

**Comment by Calinou** [comment]
> Or maybe you explicitly pick which one you want to use from the mesh menu

This is now the case since Godot 3.2.1. You can now choose between creating a *single* convex collision shape using QuickHull or *multiple* convex collision shapes using VHACD.

**Comment by jitspoe** [comment]
Ah, so all that remains is to make the "convcol*" mesh tags use the QuickHull and other ones use VHACD?  Seems like that would be better than using triangle collision and would let people explicitly choose without adding new mesh name tags, but maybe there's an argument for still supporting triangle mesh collision?

**Comment by Calinou** [comment]
@jitspoe The other mesh tags such as `-col` and `-colonly` create trimesh collision shapes, which are slower than VHACD-generated convex shapes but more accurate. There are plenty of valid use cases for them, so these shouldn't go away :slightly_smiling_face:

**Comment by jitspoe** [comment]
Sort of related: I noticed that the non-VHACD method for creating a convex collision shape adds duplicate vertices: https://github.com/godotengine/godot/issues/41522

**Comment by jitspoe** [comment]
I just did a test with a simple mesh and the VHACD "simplified" collision actually generated MORE complex collision than using the mesh directly.  Still in favor of using the non-VHACD collision for -convcolonly objects.
![image](https://user-images.githubusercontent.com/7751182/131199496-178b0a89-cefe-4e14-96bc-3d46a27311d2.png)

[boss_robot_head_collision.zip](https://github.com/godotengine/godot/files/7069843/boss_robot_head_collision.zip)

**Comment by fire** [comment]
@lyuma had trouble with generate convex decomposition a few days ago.

**Comment by jitspoe** [comment]
Using vanilla Godot for a game jam and tried to generate collision and forgot how bad this was... should really just go back to the old QuickHull method by default in my opinion with a separate -convcoldecomponly or -convcolsimplifiedonly or something for the simplifying algorithm.  It just seems too broken to be of any use right now, at least for what I've tried it for.

**Comment by fire** [comment]
I prefer COACD, but the dependencies are fairly deep. I looked into this a few years ago.
